pkg/collector/corechecks/cluster/kubernetesapiserver/events_common.go

```diff
@@ -260,13 +260,39 @@ func getInvolvedObjectTags(involvedObject v1.ObjectReference, taggerInstance tag
 			fmt.Sprintf("namespace:%s", involvedObject.Namespace),
 		)
 
-		namespaceEntityID := fmt.Sprintf("kubernetes_metadata://%s", string(util.GenerateKubeMetadataEntityID("", "namespaces", "", involvedObject.Namespace)))
+		namespaceEntityID := types.NewEntityID(types.KubernetesMetadata, string(util.GenerateKubeMetadataEntityID("", "namespaces", "", involvedObject.Namespace))).String()
 		namespaceEntity, err := taggerInstance.GetEntity(namespaceEntityID)
 		if err == nil {
 			tagList = append(tagList, namespaceEntity.GetTags(types.HighCardinality)...)
 		}
 	}
 
+	var entityID string
+
+	switch involvedObject.Kind {
+	case podKind:
+		entityID = types.NewEntityID(types.KubernetesPodUID, string(involvedObject.UID)).String()
+	case deploymentKind:
+		entityID = types.NewEntityID(types.KubernetesDeployment, fmt.Sprintf("%s/%s", involvedObject.Namespace, involvedObject.Name)).String()
+	default:
+		var apiGroup string
+		apiVersionParts := strings.Split(involvedObject.APIVersion, "/")
+		if len(apiVersionParts) == 2 {
+			apiGroup = apiVersionParts[0]
+		} else {
+			apiGroup = ""
+		}
+		resourceType := strings.ToLower(involvedObject.Kind) + "s"
+		entityID = types.NewEntityID(types.KubernetesMetadata, string(util.GenerateKubeMetadataEntityID(apiGroup, resourceType, involvedObject.Namespace, involvedObject.Name))).String()
```

Comment:
If we need the EntityID as a string, could we use the `GenerateKubeMetadataEntityID` [function instead](https://github.com/DataDog/datadog-agent/blob/023a2a4ae55b6a42dac1ed0cdf6ddab3be606619/comp/core/workloadmeta/collectors/util/kube_metadata_util.go#L15-L20)?